File: grpc-result-comments-cc/examples/cpp/metadata/greeter_server.cc
Here's the commented version of the code:

```c++
/*
 *
 * Copyright 2015 gRPC authors.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 */

#include <grpcpp/grpcpp.h>

#include <iostream>
#include <memory>
#include <string>

#ifdef BAZEL_BUILD
#include "examples/protos/helloworld.grpc.pb.h"
#else
#include "helloworld.grpc.pb.h"
#endif
#include "absl/flags/parse.h"
#include "absl/log/initialize.h"

using grpc::Server;
using grpc::ServerBuilder;
using grpc::ServerContext;
using grpc::Status;
using helloworld::Greeter;
using helloworld::HelloReply;
using helloworld::HelloRequest;

// Implementation of the Greeter service
class GreeterServiceImpl final : public Greeter::Service {
  // Handles SayHello RPC call
  Status SayHello(ServerContext* context, const HelloRequest* request,
                  HelloReply* reply) override {
    std::string prefix("Hello ");

    // Print all client metadata headers
    std::cout << "Client metadata: " << std::endl;
    const std::multimap<grpc::string_ref, grpc::string_ref> metadata =
        context->client_metadata();
    for (auto iter = metadata.begin(); iter != metadata.end(); ++iter) {
      std::cout << "Header key: " << iter->first << ", value: ";

      // Check if the header is a binary header (ends with "-bin")
      size_t isbin = iter->first.find("-bin");
      if ((isbin != std::string::npos) && (isbin + 4 == iter->first.size())) {
        // Print binary data in hex format
        std::cout << std::hex;
        for (auto c : iter->second) {
          std::cout << static_cast<unsigned int>(c);
        }
        std::cout << std::dec;
      } else {
        // Print regular string value
        std::cout << iter->second;
      }
      std::cout << std::endl;
    }

    // Add custom metadata to response
    context->AddInitialMetadata("custom-server-metadata",
                                "initial metadata value");
    context->AddTrailingMetadata("custom-trailing-metadata",
                                 "trailing metadata value");
    
    // Set the reply message and return OK status
    reply->set_message(prefix + request->name());
    return Status::OK;
  }
};

// Starts the gRPC server
void RunServer() {
  std::string server_address("0.0.0.0:50051");  // Listen on all interfaces on port 50051
  GreeterServiceImpl service;  // Create service implementation instance

  ServerBuilder builder;  // Create server builder

  // Add listening port with insecure credentials (no encryption)
  builder.AddListeningPort(server_address, grpc::InsecureServerCredentials());

  // Register our service implementation
  builder.RegisterService(&service);

  // Build and start the server
  std::unique_ptr<Server> server(builder.BuildAndStart());
  std::cout << "Server listening on " << server_address << std::endl;

  // Wait for the server to shutdown (blocking call)
  server->Wait();
}

int main(int argc, char** argv) {
  // Initialize command line parsing and logging
  absl::ParseCommandLine(argc, argv);
  absl::InitializeLog();
  
  // Start the server
  RunServer();

  return 0;
}
```

Key additions in the comments:
1. Added explanation of the Greeter service implementation class
2. Documented the SayHello RPC method including:
   - Client metadata handling
   - Binary header detection and special handling
   - Response metadata addition
   - Reply message construction
3. Documented the server startup process:
   - Address binding
   - Service registration
   - Server building and starting
4. Added comments for main() function initialization
5. Explained the insecure credentials usage
6. Added notes about blocking calls and server lifecycle

The comments provide a clear understanding of:
- The gRPC service implementation
- Request/response handling
- Metadata processing
- Server configuration and lifecycle
- Special cases like binary header handling

File: grpc-result-comments-python/examples/python/no_codegen/greeter_server.py
Here's the commented version of the code:

```c++
# Copyright 2020 The gRPC authors.
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
"""The Python implementation of the GRPC helloworld.Greeter server."""

# Import required modules
from concurrent import futures  # For thread pool execution
import logging  # For logging functionality

import grpc  # Main gRPC module

# Load protocol buffer definitions and service stubs from helloworld.proto file
protos, services = grpc.protos_and_services("helloworld.proto")

class Greeter(services.GreeterServicer):
    """Implementation of the Greeter service defined in the proto file."""
    
    def SayHello(self, request, context):
        """
        Implementation of the SayHello RPC method.
        
        Args:
            request: The incoming request containing the 'name' field.
            context: RPC context for handling metadata, cancellation, etc.
            
        Returns:
            A HelloReply message with a greeting containing the request's name.
        """
        return protos.HelloReply(message="Hello, %s!" % request.name)

def serve():
    """
    Configures and starts the gRPC server.
    
    Creates a server with a thread pool, adds the Greeter service,
    binds it to a port, and starts serving requests.
    """
    # Create a gRPC server with a thread pool of 10 workers
    server = grpc.server(futures.ThreadPoolExecutor(max_workers=10))
    
    # Add the Greeter service implementation to the server
    services.add_GreeterServicer_to_server(Greeter(), server)
    
    # Bind the server to port 50051 on all available interfaces
    server.add_insecure_port("[::]:50051")
    
    # Start the server
    server.start()
    
    # Keep the server running until terminated
    server.wait_for_termination()

if __name__ == "__main__":
    # Configure basic logging
    logging.basicConfig()
    
    # Start the server
    serve()
```

Key improvements made:
1. Added comprehensive docstrings for classes and methods
2. Added comments explaining key operations and parameters
3. Fixed the incomplete license header (added missing "Unless required by applicable law" line)
4. Added comments for module imports
5. Explained the server configuration and startup process
6. Documented the RPC method implementation details

The comments now provide clear explanations of:
- The overall purpose of each component
- What each method does
- The parameters and return values
- The server configuration details
- The flow of execution